From 0c1fdae0f0cf62ae12f5d01a1650cad710568b0a Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 14 Dec 2021 12:44:45 +0200
Subject: [PATCH] Revert "[SW-63186] gaudi2: security: set axprot for tpc"

Moving axprot configuration of tpc to zephyr post isolation.
This is required as the engines which have isolation logic must
be accessed/configured post isolation configuration.

This reverts commit aed5a22dbaeb814c8e6ccca7dcb907c2caeaf942.

Change-Id: I0d0a88e66e87e1d71d47bc016add57af7cde13ea
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index c839c75..4586686 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -40,10 +40,6 @@
 #include <arc_farm_kdma_mstr_if_core_hbw_regs.h>
 #include <arc_farm_kdma_mstr_if_core_lbw_regs.h>
 #include <psoc_global_conf_regs.h>
-#include <dcore0_tpc0_mstr_if_core_hbw_regs.h>
-#include <dcore0_tpc0_mstr_if_core_lbw_regs.h>
-#include <dcore0_tpc6_mstr_if_core_hbw_regs.h>
-#include <dcore0_tpc6_mstr_if_core_lbw_regs.h>
 #include <dcore0_mme_sbte0_mstr_if_core_hbw_regs.h>
 #include <dcore0_mme_wb0_mstr_if_core_hbw_regs.h>
 #include <dcore0_mme_ctrl_mstr_if_core_hbw_regs.h>
@@ -212,7 +208,6 @@
 	AXPROT_VDEC,
 	AXPROT_ROT,
 	AXPROT_HMMU,
-	AXPROT_TPC,
 	AXPROT_MME,
 	AXPROT_PDMA,
 };
@@ -934,10 +929,6 @@
 #define HMMU_RTR_INST_DIST	0x10000
 #define HMMU_INST_NUM		NUM_OF_DCORES
 #define HMMU_INST_DIST		DCORE_DIST
-#define TPC_RTR_INST_NUM	6
-#define TPC_RTR_INST_DIST	0x10000
-#define TPC_INST_NUM		NUM_OF_DCORES
-#define TPC_INST_DIST		DCORE_DIST
 #define MME_SBTE_RTR_INST_NUM	5
 #define MME_SBTE_RTR_INST_DIST	0x8000
 #define MME_WB_RTR_INST_NUM	2
@@ -974,14 +965,6 @@
 		.dcore_num = HMMU_INST_NUM,
 		.inst_dist = HMMU_RTR_INST_DIST,
 		.inst_num = HMMU_RTR_INST_NUM,
-	},
-	[AXPROT_TPC] = {
-		.base_reg = mmDCORE0_TPC0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
-		.value = AXPROT_USER_VAL,
-		.dcore_dist = TPC_INST_DIST,
-		.dcore_num = TPC_INST_NUM,
-		.inst_dist = TPC_RTR_INST_DIST,
-		.inst_num = TPC_RTR_INST_NUM,
 	},
 	[AXPROT_MME] = {
 		.base_reg = mmDCORE0_MME_SBTE0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
@@ -1065,26 +1048,6 @@
 	regmap_writel(PROT_OVR_USER, mmPSOC_GLOBAL_CONF_TRACE_AXPROT);
 }
 
-static void preboot_secure_tpc(void)
-{
-	/* Configure tpc to user level */
-
-	/* HBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_TPC);
-
-	axprot_reg[AXPROT_TPC].base_reg = mmDCORE0_TPC0_MSTR_IF_CORE_LBW_AXPROT_OVRD;
-
-	/* LBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_TPC);
-
-	/*
-	 * DCORE0 has an additional TPC compared to other DCOREs.
-	 */
-	regmap_writel(AXPROT_USER_VAL, mmDCORE0_TPC6_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-
-	regmap_writel(AXPROT_USER_VAL, mmDCORE0_TPC6_MSTR_IF_CORE_LBW_AXPROT_OVRD);
-}
-
 static void preboot_secure_mme(void)
 {
 	/* Configure mme to user level */
@@ -1139,7 +1102,6 @@
 		preboot_secure_pmmu();
 		preboot_secure_arc_farm();
 		preboot_secure_trace();
-		preboot_secure_tpc();
 		preboot_secure_mme();
 		preboot_secure_pdma();
 		preboot_secure_lbw();
